ENIGMAsystem/SHELL/Widget_Systems/TinyFileDialogs/dialogs.cpp

```diff
@@ -361,7 +361,27 @@ namespace enigma_user
 
     const char *path = tinyfd_selectFolderDialog("Select Directory", dname.c_str());
 
-    return path ? : "";
+    bool ok_pressed = true;
+
+    if (path == NULL)
+      ok_pressed = false;
+
+    static string str_result;
+    
+    if (ok_pressed == true)
+    {
+      char cstr_path[PATH_MAX];
+      strncpy(cstr_path, path, PATH_MAX);
+
+      char *slash = (char *)"/";
+      strcat(cstr_path, slash);
```

Comment:
Prefer `char slash[] = "/"`, if you need this to be non-const char. Also, this `strcat` does not respect the buffer size; you would need to use `strncat` or add `1` to `PATH_MAX` in the boundary above.